File: Java/17-pattern-alpha-hill.java
import java.util.Scanner;

public class Solution {
    public static void alphaHill(int n) {
        // Write your code here
        int i, j;
        for(i = 0; i<n; i++){
            for(j = 0; j<n-i-1; j++)
                System.out.print(" ");
            
            char ch = 'A';
            int middle = (2 * i + 1) / 2;
            for(j = 1; j<=2*i+1; j++){
                System.out.print(ch + " ");
                if(j<=middle)
                    ch++;
                else
                    ch--;
            }

            for(j = 0; j<n-i-1; j++)
                System.out.print(" ");
            
            System.out.println();
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        alphaHill(n);
        scanner.close();
    }
}

File: Python/17-pattern-alpha-hill.py
def alphaHill(n):
    for i in range(n):
        for j in range(n - i - 1):
            print(" ", end="")
        
        ch = 'A'
        middle = (2 * i + 1) // 2
        for j in range(1, 2 * i + 2):
            print(ch, end=" ")
            if j <= middle:
                ch = chr(ord(ch) + 1)
            else:
                ch = chr(ord(ch) - 1)
        
        for j in range(n - i - 1):
            print(" ", end="")
        
        print()

n = int(input())
alphaHill(n)
